### ISSUE TITLE
Take elevation into account when determining if a token is within detection mode range

### ISSUE THREAD

**Issue by Fyorl** [foundryvtt/foundryvtt#8391]
Originally reported by Eserg#9576 https://discord.com/channels/170995199584108546/956306859491471420/1031234217553702942

**ALL MODULES DISABLED?** True  
**OS, Hosting, Browser (if applicable):** Windows 10, Self, Foundry 10.288 (Native/Electron)  
**Short Description of bug:** Detection modes don't calculate height.  
**Simple steps to reproduce the bug:** Change another token elevation. Apply to another token "Hearing" or "Sence All" Detection Mode  
**Screenshots and/or console errors:** In the screenshot, the token is set to "Hearing" detection mode at 6ft, it can hear a creature that is approximately 62ft away from it 

![image](https://user-images.githubusercontent.com/207433/196168034-1992f37c-3eb3-47d4-93d0-bef9074f2418.png)

**Comment by Fyorl** [comment]
This was raised as a bug report but I'm reasonably sure it's working as designed and we don't attempt to calculate this at all, so I've left it as a potential feature request/improvement.

**Comment by aaclayton** [comment]
> This was raised as a bug report but I'm reasonably sure it's working as designed and we don't attempt to calculate this at all, so I've left it as a potential feature request/improvement.

That's correct - this would be a feature expansion that would be almost certainly be a part of https://github.com/foundryvtt/foundryvtt/issues/4623 rather than a stand-alone change.
